File: Easy/350-intersection-of-two-arrays-ii/try-7.py
/*
 * Problem #350: Intersection of Two Arrays II
 * Difficulty: Easy
 * Submission: Try 7
 * status: Accepted
 * Language: python3
 * Date: 04/12/2024, 21:56:47
 * Link: https://leetcode.com/problems/intersection-of-two-arrays-ii/
 */

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        d={}
        l1=[]
        for i in nums1:
            if i not in d:
                d[i]=1
            else:
                d[i]+=1
        
        for i in nums2:
            if i in d and d[i]>0:
                l1.append(i)
                d[i]=d[i]-1
        return l1
        
        
        

File: Medium/238-product-of-array-except-self/try-7.cpp
/*
 * Problem #238: Product of Array Except Self
 * Difficulty: Medium
 * Submission: Try 7
 * status: Accepted
 * Language: cpp
 * Date: 14/12/2025, 19:59:50
 * Link: https://leetcode.com/problems/product-of-array-except-self/
 */

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> pre(nums.size(),1), suf(nums.size(), 1);
        vector<int> result;
        for(int i = 1;i < nums.size() ;i++){
            pre[i] *= pre[i-1]*nums[i-1];
            suf[nums.size() - 1 - i] *= suf[nums.size() - i] * nums[nums.size() - i];
        }
        for(int i = 0;i < nums.size();i++){
            result.push_back(pre[i] * suf[i]);
        }
        return result;

    }
};
